lib/core/all/hash.rbi

```diff
@@ -0,0 +1,337 @@
+    .returns(T::Hash[K, V])
+  end
+  sig {returns(Enumerator[[K, V]])}
+  def each(&blk); end
+
+  sig do
+    params(
+        blk: T.proc.params(arg0: K).returns(BasicObject),
+    )
+    .returns(T::Hash[K, V])
+  end
+  sig {returns(Enumerator[[K, V]])}
+  def each_key(&blk); end
+
+  sig do
+    params(
+        blk: T.proc.params(arg0: K, arg1: V).returns(BasicObject),
+    )
+    .returns(T::Hash[K, V])
+  end
+  sig {returns(Enumerator[[K, V]])}
+  def each_pair(&blk); end
+
+  sig do
+    params(
+        blk: T.proc.params(arg0: V).returns(BasicObject),
+    )
+    .returns(T::Hash[K, V])
+  end
+  sig {returns(Enumerator[[K, V]])}
+  def each_value(&blk); end
+
+  sig {returns(T::Boolean)}
+  def empty?(); end
+
+  sig do
+    params(
+        arg0: K,
+    )
+    .returns(V)
+  end
+  sig do
+   type_parameters(:X).params(
+      arg0: K,
+      arg1: T.type_parameter(:X),
+    )
+    .returns(T.any(V, T.type_parameter(:X)))
+  end
+  sig do
+   type_parameters(:X).params(
+        arg0: K,
+        blk: T.proc.params(arg0: K).returns(T.type_parameter(:X)),
+    )
+    .returns(T.any(V, T.type_parameter(:X)))
+  end
+  def fetch(arg0, arg1=T.unsafe(nil), &blk); end
+
+  sig do
+    params(
+        arg0: K,
+    )
+    .returns(T::Boolean)
+  end
+  def has_key?(arg0); end
+
+  sig do
+    params(
+        arg0: V,
+    )
+    .returns(T::Boolean)
+  end
+  def has_value?(arg0); end
+
+  sig {returns(Hash)}
+  sig do
+    params(
+        default: Object,
+    )
+    .returns(Hash)
+  end
+  sig do
+    params(
+        blk: T.proc.params(hash: Hash, key: Object).returns(Object)
+    )
+    .void
+  end
+  def initialize(default=T.unsafe(nil), &blk); end
+
+  sig {returns(String)}
+  def inspect(); end
+
+  sig {returns(T::Hash[V, K])}
+  def invert(); end
+
+  sig do
+    params(
+        blk: T.proc.params(arg0: K, arg1: V).returns(BasicObject),
+    )
+    .returns(T::Hash[K, V])
+  end
+  sig {returns(Enumerator[[K, V]])}
+  def keep_if(&blk); end
+
+  sig do
+    params(
+        arg0: V,
+    )
+    .returns(T.nilable(K))
+  end
+  def key(arg0); end
+
+  sig do
+    params(
+        arg0: K,
+    )
+    .returns(T::Boolean)
+  end
+  def key?(arg0); end
+
+  sig {returns(T::Array[K])}
+  def keys(); end
+
+  sig {returns(Integer)}
+  def length(); end
+
+  sig do
+    params(
+        arg0: K,
+    )
+    .returns(T::Boolean)
+  end
+  def member?(arg0); end
+
+  sig do
+    type_parameters(:A ,:B).params(
+        arg0: T::Hash[T.type_parameter(:A), T.type_parameter(:B)],
+    )
+    .returns(T::Hash[T.any(T.type_parameter(:A), K), T.any(T.type_parameter(:B), V)])
+  end
+  sig do
+    type_parameters(:A ,:B).params(
+        arg0: T::Hash[T.type_parameter(:A), T.type_parameter(:B)],
+        blk: T.proc.params(arg0: K, arg1: V, arg2: T.type_parameter(:B)).returns(T.any(V, T.type_parameter(:B))),
+    )
+    .returns(T::Hash[T.any(T.type_parameter(:A), K), T.any(T.type_parameter(:B), V)])
+  end
+  def merge(arg0, &blk); end
```

Comment:
Since `Hash#merge` is [variadic as of 2.6](https://ruby-doc.org/core-2.6.3/Hash.html#method-i-merge), it'd be nice to support the new signature:

```ruby
  sig do
    type_parameters(:A, :B).params(
        args: T::Hash[T.type_parameter(:A), T.type_parameter(:B)],
    )
    .returns(T::Hash[T.any(T.type_parameter(:A), K), T.any(T.type_parameter(:B), V)])
  end
  sig do
    type_parameters(:A, :B).params(
        args: T::Hash[T.type_parameter(:A), T.type_parameter(:B)],
        blk: T.proc.params(arg0: K, arg1: V, arg2: T.type_parameter(:B)).returns(T.any(V, T.type_parameter(:B))),
    )
    .returns(T::Hash[T.any(T.type_parameter(:A), K), T.any(T.type_parameter(:B), V)])
  end
  def merge(*args, &blk); end
```